src/sentry/search/events/fields.py

```diff
@@ -2547,6 +2599,149 @@ def _resolve_unimplemented_alias(self, alias: str) -> SelectType:
         """
         raise NotImplementedError(f"{alias} not implemented in snql field parsing yet")
 
+    def _private_project_threshold_multi_function(self) -> SelectType:
+        lcp_index = Function(
+            "indexOf", [self.column("measurements_key"), "lcp"], MEASUREMENTS_LCP_INDEX_ALIAS
+        )
+
+        return Function(
+            "multiIf",
+            [
+                Function(
+                    "equals",
+                    [
+                        Function(
+                            "tupleElement",
+                            [self.resolve_field("project_threshold_config"), 1],
+                        ),
+                        "lcp",
+                    ],
+                ),
+                Function(
+                    "if",
+                    [
+                        Function("equals", [lcp_index, 0]),
+                        None,
+                        Function(
+                            "arrayElement",
+                            [
+                                self.column("measurements_value"),
+                                lcp_index,
+                            ],
+                        ),
+                    ],
+                ),
```

Comment:
changed this a bit- we were previously checking if measurements_key first has "lcp" and if it does, get the index.
changed it to computing lcp index in measurements_key and defaulting to None if index is 0.

Reply:
ooh good catch, let me try

Reply:
nice, that worked! it's so much nicer now